lab/SoundTest/example_scores.js

```diff
@@ -0,0 +1,156 @@
+  map_pitch,
+  Melody,
+  MusicLoop,
+  Note,
+  parse_cycle,
+  parse_melody,
+  Rest,
+  Score,
+} from "../lablib/music/Score.js";
+import { transpose_scale_degree } from "../lablib/music/transpose.js";
+import { Rational } from "../lablib/Rational.js";
+
+/**
+ * Convert a scale to a pitch, using a fixed octave
+ * @param {number[]} scale Array of pitch classes for the scale
+ * @param {number} octave octave number
+ * @returns {function(number):number} A callback function for transforming
+ * scale degrees to pitches
+ */
+function scale_to_pitch(scale, octave) {
+  return (scale_degree) => {
+    const pitch_class = scale[scale_degree];
+    return MidiPitch.from_pitch_octave(pitch_class, octave);
+  };
+}
+
+const SCALE = [C, E, F, G, GS, B];
+const SCALE3 = scale_to_pitch(SCALE, 3);
+const SCALE4 = scale_to_pitch(SCALE, 4);
+const SCALE5 = scale_to_pitch(SCALE, 5);
+
+export function layered_melody() {
+  const pedal = parse_melody([C3, N4], [REST, N4]);
+
+  const N4D = new Rational(3, 8);
+
+  // scores are expressed in scale degrees then converted to pitches
+  const scale_arp = map_pitch(
+    SCALE4,
+    parse_melody(
+      // Measure 1
+      [0, N4],
+      [1, N4],
+      [2, N4],
+      [3, N4],
+      // Measure 2
+      [4, N4D],
+      [REST, N8],
+      [5, N4D],
+      [REST, N8]
+    )
+  );
+
+  const cycle_length = N1;
+  const cycle_a = map_pitch(
+    SCALE5,
+    parse_cycle(cycle_length, [0, REST, 1, 2, REST, 4])
+  );
+
+  const cycle_b = map_pitch(
+    SCALE5,
+    parse_cycle(cycle_length, [
+      [0, 3],
+      [5, REST],
+      [0, 4, 2, 4],
+    ])
+  );
+
+  const sine_part = new MusicLoop(new Rational(34, 1), pedal);
+  const square_part = new Melody(
+    new Rest(new Rational(2, 1)),
+    new MusicLoop(new Rational(22, 1), scale_arp)
+  );
+  const poly_part = new Harmony(
+    new Melody(
+      new Rest(new Rational(8, 1)),
+      new MusicLoop(new Rational(12, 1), cycle_a),
+      new Rest(new Rational(4, 1)),
+      new MusicLoop(new Rational(4, 1), cycle_a)
+    ),
+    new Melody(
+      new Rest(new Rational(8, 1)),
+      new MusicLoop(new Rational(8, 1), cycle_b),
+      new Rest(new Rational(4, 1)),
+      new MusicLoop(new Rational(4, 1), cycle_b),
+      new Rest(new Rational(4, 1)),
+      new MusicLoop(new Rational(4, 1), cycle_b)
+    )
+  );
+
+  return new Score(
+    ["sine", sine_part],
+    ["square", square_part],
+    ["poly", poly_part]
+  );
+}
+
+export function phase_scale() {
+  // Three scales initially the same, but with different lengths
+  const phase_a = parse_cycle(new Rational(6, 8), [0, 1, 2, 3, 2, 1]);
+  const phase_b = parse_cycle(new Rational(1, 1), [0, 1, 2, 3, 4, 3, 2, 1]);
+  const phase_c = parse_cycle(
+    new Rational(10, 8),
+    [0, 1, 2, 3, 4, 5, 4, 3, 2, 1]
+  );
+
+  // lcm(6, 8, 10) = 120, divide by 4 to get number of measures, 30
+  const phase_part_scale = new MusicLoop(
+    new Rational(30, 1),
+    new Harmony(phase_a, phase_b, phase_c)
+  );
```

Comment:
This should be a harmony of loops rather than a loop of harmony. There is a difference in playback!